File: Loadshop.DomainServices/Loadshop/Services/Data/SpecialInstructionData.cs
using System;
using System.Collections.Generic;

namespace Loadshop.DomainServices.Loadshop.Services.Data
{
    public class SpecialInstructionData
    {
        public long SpecialInstructionId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Comments { get; set; }
        public Guid CustomerId { get; set; }
        public string OriginAddress1 { get; set; }
        public string OriginCity { get; set; }
        public string OriginState { get; set; }
        public string OriginPostalCode { get; set; }
        public string OriginCountry { get; set; }
        public string DestinationAddress1 { get; set; }
        public string DestinationCity { get; set; }
        public string DestinationState { get; set; }
        public string DestinationPostalCode { get; set; }
        public string DestinationCountry { get; set; }
        public List<SpecialInstructionEquipmentData> SpecialInstructionEquipment { get; set; } = new List<SpecialInstructionEquipmentData>();
    }
}

File: Loadshop.Web.UI/src/app/shared/models/special-instruction-data.ts
import { SpecialInstructionEquipmentData } from './special-instruction-equipment-data';

export interface SpecialInstructionData {
  specialInstructionId: number;
  name: string;
  description: string;
  comments: string;
  customerId: string;
  originAddress1: string;
  originCity: string;
  originState: string;
  originPostalCode: string;
  originCountry: string;
  destinationAddress1: string;
  destinationCity: string;
  destinationState: string;
  destinationPostalCode: string;
  destinationCountry: string;
  specialInstructionEquipment: SpecialInstructionEquipmentData[];
}

export const defaultSpecialInstructions: SpecialInstructionData = {
  specialInstructionId: null,
  name: null,
  description: null,
  comments: null,
  customerId: null,
  originAddress1: null,
  originCity: null,
  originState: null,
  originPostalCode: null,
  originCountry: null,
  destinationAddress1: null,
  destinationCity: null,
  destinationState: null,
  destinationPostalCode: null,
  destinationCountry: null,
  specialInstructionEquipment: [],
};
